### PR TITLE
fix type issues in datastore

### PR DESCRIPTION
**Pull request by jzxcd**

### CODE REVIEW

File: perfeed/llms/base_client.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
 
-
+@dataclass
```
Comment (chihangwang) [inline perfeed/llms/base_client.py]: It's not a dataclass, but an interface. Please remove the decorator

File: perfeed/models/pr_summary.py [hunk 2]
```diff
@@ -75,5 +75,5 @@ class PRSummaryMetadata(BaseModel):
     llm_provider: str
     model: str
     pr_created_at: str
-    pr_merged_at: str
+    pr_merged_at: Optional[str] = None
```
Comment (chihangwang) [inline perfeed/models/pr_summary.py:78]: `pr_merged_at: str | None = None` would work without additional import for `Optional`

**Review by chihangwang — APPROVED** [review]
LGTM, but please remove `dataclass` decorator

File: perfeed/models/pr_summary.py (hunk 2 above)
Comment (jzxcd) [reply perfeed/models/pr_summary.py:78]: i will keep as is just for the consistency between this and PullRequest.merged_at

**Pull request merged**